File: backend/src/main/java/at/ac/tuwien/sepm/groupphase/backend/endpoint/dto/TicketDto.java
package at.ac.tuwien.sepm.groupphase.backend.endpoint.dto;

import at.ac.tuwien.sepm.groupphase.backend.entity.*;
import lombok.Data;
import lombok.EqualsAndHashCode;

import javax.persistence.EnumType;
import javax.persistence.Enumerated;

@Data
@EqualsAndHashCode(callSuper=false)
public class TicketDto {



    private Long id;


    private Order customerOrder;


    private EventPerformance performance;


    private Seat seat;

    private Float price;

    @Enumerated(EnumType.STRING)
    private Ticket.Status status;

}

File: frontend/src/app/dtos/ticket-dto.ts
import { Order } from './order';
import { EventPerformance } from './event-performance';
import { Seat } from './seat';

export class TicketDto {

    constructor(
        public id: number,
        public customerOrder: Order,
        public performance:EventPerformance,
        public seat: Seat,
        public status: string,
        public price: number,) {
            
            
    }
}
